src/backend/TrafficCourts/Workflow.Service/Consumers/DisputeUpdateRequestAcceptedConsumer.cs

```diff
@@ -116,7 +120,17 @@ public async Task Consume(ConsumeContext<DisputeUpdateRequestAccepted> context)
                         }
                         break;
                     case DisputeUpdateRequestUpdateType.DISPUTANT_DOCUMENT:
-                        // TODO: update document metadata set StaffReviewStatus to Accepted
+                        // update all documents in this patch to ACCEPTED
+                        if (patch.UploadedDocuments is not null) {
+                            foreach (UploadDocumentRequest uploadDocumentRequest in patch.UploadedDocuments) {
+                                Guid? documentId = uploadDocumentRequest.DocumentId;
+                                if (documentId is not null) {                                    
+                                    Dictionary<string, string> properties = new();
+                                    properties.Add(DocumentProperties.PropertyName.StaffReviewStatus, DisputeUpdateRequestStatus.ACCEPTED.ToString());

```

Comment:
That's how I had it originally, but Phil correctly pointed out that this additionally retrieves the entire file when retrieving the document properties which is deemed unnecessary.
This was supposed to just change the one document property, but more digging shows that this is not case and this does in fact wipe out all doc properties.  Argh. Guess we're back to pulling the entire file from COMS just to update a single tag value.

Reply:
There is the ability to fetch the tags for a object, [FetchTagsAsync](https://github.com/bcgov/jag-traffic-courts-online/blob/f45ce9b89842b61f388ff3ab3c52ebb401a14b3a/src/backend/TrafficCourts/TrafficCourts.Coms.Client/Monitoring/InstrumentedObjectManagementClient.cs#L123). So if we need to fetch all the tags, manipulate the collection and re-save that should be technically possible.

The underlying call is this [/object/{objectId}/tagging](https://github.com/bcgov/common-object-management-service/blob/9d1c6ea9f1ad84ad47e5835a4ad399c2c1cd612b/app/src/docs/v1.api-spec.yaml#L641). This endpoint says it adds tags, but doesnt say it replaces it. These are the kinds of issues I have, needed to read the source.  

But for now, I guess can pull the document down to change it since we are short on time. Do we expect this function to occur a lot. Just makes it slow to fetch possibly megabytes of data to and from S3 just to change tags.

Reply:
Yeah, it would be nice, and the better approach, to only update the tags. For now I reverted back to fetching the entire file with tags to update a tag value. Also definitely not ideal this does work and does not clear out virus scan properties.